src/algebraic_geometry/prime_spectrum.lean

```diff
@@ -261,6 +261,11 @@ lemma zero_locus_Union {ι : Sort*} (s : ι → set R) :
   zero_locus (⋃ i, s i) = (⋂ i, zero_locus (s i)) :=
 (gc_set R).l_supr
 
+lemma zero_locus_bUnion {s : set R} :
+  zero_locus (⋃ i ∈ s, {i} : set R) = ⋂ i ∈ s, zero_locus {i} :=
```

Comment:
Does this generalize to arbitrary `f i` instead of just `f I = {i}`?

Reply:
I think `(gc_set R).l_Sup` is the proof?

Reply:
I am not sure that I know how to do this.  Mathematically, the fact that I am breaking `s` into its singletons is not important: any way of writing `s` as a union of (not necessarily disjoint) subsets will work.  What seems to be important for me, though, is that the union is over "members" of something.  I have not been able to formalize this statement.  If you can give me a hint, I can try to make it work!  Thanks!

Reply:
> I think (gc_set R).l_Sup is the proof?

I have tried it and it does not work: did you change the statement of the lemma?

Reply:
```
lemma zero_locus_bUnion (s : set (set R)) :
  zero_locus (⋃ s' ∈ s, s' : set R) = ⋂ s' ∈ s, zero_locus s' :=
by rw [set.bUnion_eq_Union, zero_locus_Union, set.bInter_eq_Inter]
```
@adomani what do you think of :arrow_up: ?

Reply:
I like this!  I have tried it out on the PR where I need it and it seems to work in this form as well.

Thanks!